subscription.go

```diff
@@ -31,3 +52,65 @@ func (sub *Subscription) Next(ctx context.Context) (*Message, error) {
 func (sub *Subscription) Cancel() {
 	sub.cancelCh <- sub
 }
+
+func (sub *Subscription) close() {
+	close(sub.ch)
+}
+
+func (sub *Subscription) sendNotification(evt PeerEvent) {
+	sub.backlogMx.Lock()
+	defer sub.backlogMx.Unlock()
+
+	sub.addToBacklog(evt)
+
+	select {
+	case sub.backlogCh <- struct{}{}:
+	default:
```

Comment:
uber nit: I'd just move this to the `addToBacklog` function. We shouldn't call it  in the `else` case anyways.

Reply:
which `else` case? 

I left them separate mostly as a strategy for separating out the code components so we could play around with a few options. However, given that `addToBacklog` is only called once, and it's by `sendNotification` if we wanted to we could just combine these functions. Having them separate is only useful if it helps with code readability.

Reply:
https://github.com/libp2p/go-libp2p-pubsub/pull/190/files#diff-384bf0d04fad588e8589bdda0252018bR78